theano/sandbox/cuda/dnn.py

```diff
@@ -1544,20 +1554,41 @@ def c_init_code_struct(self, node, name, sub):
                "(out): %%s", cudnnGetErrorString(err%(name)s));
   %(fail)s
 }
+
+if ((err%(name)s = cudnnCreatePoolingDescriptor(&pool%(name)s)) != CUDNN_STATUS_SUCCESS) {
+  PyErr_Format(PyExc_MemoryError, "could not allocate pooling "
+                "descriptor: %%s", cudnnGetErrorString(err%(name)s));
+  %(fail)s
+}
 """ % dict(name=name, fail=sub['fail'])
 
     def c_cleanup_code_struct(self, node, name):
         return """
 if (input%(name)s != NULL) { cudnnDestroyTensorDescriptor(input%(name)s); }
 if (output%(name)s != NULL) { cudnnDestroyTensorDescriptor(output%(name)s); }
+if (pool%(name)s != NULL) { cudnnDestroyPoolingDescriptor(pool%(name)s); }
 """ % dict(name=name)
 
     def c_code(self, node, name, inputs, outputs, sub):
-        desc = inputs[1]
+        ws = inputs[1]
+        stride = inputs[2]
+        pad = inputs[3]
```

Comment:
This change the interface. If user have pickled graph, it won't work. They will get a strange error.

I'm not sure what to do. I see 2 options:

1) if len(inputs) != 2, give a better error message like, "Theano DnnPool internal changed, rebuild your graph"
2) You could try to get the data from the descriptor

2) is better as it allow to unpickle old graph. Can you try to do it? I think it isn't very hard. To test it, pickle a small graph with the master of Theano that have both pool op in it. Then unpickle it and compile it with an empty cache.